### PR TITLE
Update version 3 definition docs

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME
docs/definition.rst

### CODE REVIEW

File: docs/definition.rst [hunk 1]
```diff
@@ -69,7 +73,7 @@ Here is a sample version 3 EE file. To use Ansible Builder 3.x, you must specify
         - RUN $PKGMGR module enable postgresql:15 -y
         - RUN $PKGMGR install -y postgresql
       prepend_galaxy:
-        - ADD _build/configs/ansible.cfg ~/.ansible.cfg
+        - ADD _build/configs/ansible.cfg /runner/.ansible.cfg
```
Comment (Shrews) [inline docs/definition.rst]: The example base image actually has `/home/runner` set as `$HOME`, but perhaps we can avoid the whole "knowing what $HOME is in the selected base image" issue by just using `/etc/ansible/ansible.cfg` in the example?

**Review by Shrews — CHANGES_REQUESTED** [review]
Other things that I noticed are wrong with this example (if you could fix them, too):

- The selected base image has Python 3.8, so `ansible-core==2.14.4` is not going to work since that requires 3.9. A `2.13` version should work though.
- The selected base image uses `microdnf`, so we need to set that in `options.package_manager_path`.

File: docs/definition.rst [hunk 2]
```diff
@@ -59,17 +59,22 @@ Here is a sample version 3 EE file. To use Ansible Builder 3.x, you must specify
...
+    # options:
+    #  package_manager_path: /usr/bin/microdnf
+
     additional_build_files:
         - src: files/ansible.cfg
           dest: configs
 
     additional_build_steps:
       prepend_base:
+        - RUN echo This is a preprend base command!
```
Comment (Shrews) [inline docs/definition.rst]:
```suggestion
        - RUN echo This is a prepend_base command!
```

File: docs/definition.rst [hunk 3]
```diff
@@ -59,17 +59,22 @@ Here is a sample version 3 EE file. To use Ansible Builder 3.x, you must specify
...
         # Enable Non-default stream before packages provided by it can be installed. (optional)
-        - RUN $PKGMGR module enable postgresql:15 -y
-        - RUN $PKGMGR install -y postgresql
+        # - RUN $PKGMGR module enable postgresql:15 -y
```
Comment (Shrews) [inline docs/definition.rst:74]: Why comment these out?

Comment (Akasurde) [reply docs/definition.rst:74]: Postgresql module stream is available in `ee-minimal-rhel8` or downstream images. Also, this is an optional step there is no actual use of installing postgresql.

File: docs/definition.rst (hunk 2 above)
Comment (Akasurde) [reply docs/definition.rst]: Thanks

**Review by Shrews — APPROVED** [review]

**Pull request merged**